app/src/main/kotlin/com/twain/interprep/presentation/ui/components/InterviewCard.kt

```diff
@@ -0,0 +1,206 @@
+    company = "Uber",
+    interviewType = "In-person",
+    role = "Software Engineer",
+    roundNum = 2,
+    jobPostLink = "https://www.linkedin.com/jobs/collections/recommended/?currentJobId=3512066424",
+    companyLink = "https://www.uber.com/ca/en/ride/",
+    interviewer = "John Smith",
+    interviewStatus = InterviewStatus.NEXTROUND
+)
+
+@Composable
+@Preview
+fun UpcomingInterviewCard() {
+    calendar.set(Calendar.MONTH, Calendar.AUGUST)
+    calendar.set(Calendar.YEAR, 2022)
+    InterviewCard(
+        interview = interview,
+        onClick = {},
+        color = InterviewCardColor.UpcomingInterviewCardColor
+    )
+}
+
+@Composable
+@Preview
+fun ComingNextInterviewCard() {
+    calendar.set(Calendar.MONTH, Calendar.AUGUST)
+    calendar.set(Calendar.YEAR, 2022)
+    InterviewCard(
+        interview = interview,
+        onClick = {},
+        color = InterviewCardColor.ComingNextInterviewColor
+    )
+}
+
+@Composable
+@Preview
+fun PastInterviewCard() {
+    calendar.set(Calendar.MONTH, Calendar.AUGUST)
+    calendar.set(Calendar.YEAR, 2022)
+    InterviewCard(
+        interview = interview,
+        onClick = {},
+        color = InterviewCardColor.PastInterviewCardColor
+    )
+}
+
+@Composable
+fun InterviewCard(
+    interview: Interview,
+    onClick: () -> Unit,
+    color: InterviewCardColor,
+    height: Dp = 106.dp
+) {
+    val containerColor: Color
+    val dateBoxColor: Color
+    val textColor: Color
+    when (color) {
+        is InterviewCardColor.UpcomingInterviewCardColor -> {
+            containerColor = LightPurple
+            dateBoxColor = Primary30
+            textColor = MaterialTheme.colorScheme.surfaceTint
+        }
+        is InterviewCardColor.ComingNextInterviewColor -> {
+            containerColor = LightGreen
+            dateBoxColor = Green
+            textColor = BrightGreen
+        }
+        is InterviewCardColor.PastInterviewCardColor -> {
+            containerColor = Gray
+            dateBoxColor = DarkGray
+            textColor = LightGray
+        }
+    }
+    ElevatedCard(
+        shape = Shapes.large,
+        colors = CardDefaults.cardColors(containerColor = containerColor),
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable(onClick = onClick)
+            .height(height),
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(16.dp)
+        ) {
+            Box(
+                modifier = Modifier
+                    .size(74.dp)
+                    .clip(Shapes.large)
+                    .background(dateBoxColor),
+            ) {
+                Column(
+                    modifier = Modifier.fillMaxSize(),
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    verticalArrangement = Arrangement.Center
+                ) {
+                    Text(
+                        text = SimpleDateFormat("dd").format(interview.date.time),
```

Comment:
I didn't find a better way to format timestamps. I searched around and used `SimpleDateFormat` with `Calendar` object to format the time.